tests/test_queryselector.py

```diff
@@ -0,0 +1,301 @@
+Page.$$eval()   Page.querySelectorAllEval() Page.JJeval()
+
+"""
+
+import pytest
+
+from syncer import sync
+
+from pyppeteer.errors import ElementHandleError
+from tests.conftest import chrome_only
+
+
+@chrome_only
+@sync
+async def test_page_Jeval_executes_js_func(isolated_page):
+    """Test Page().Jeval() aka `querySelectorEval()` method,
+    which executes JS function with an element which matches ``selector``
+
+    """
+    page = isolated_page
+    await page.setContent('<section id="testAttribute">43543</section>')
+    idAttribute_Jeval = await page.Jeval('section', "e => e.id")
+    idAttribute_query_eval = await page.querySelectorEval('section', "e => e.id")
+    assert idAttribute_Jeval == idAttribute_query_eval == 'testAttribute'
+
+
+@chrome_only
+@sync
+async def test_Jeval_accepts_args(isolated_page):
+    """Test Page().Jeval() should accept arguments."""
+    page = isolated_page
+    await page.setContent('<section>hello</section>')
+    text = await page.Jeval('section', "(e, suffix) => e.textContent + suffix", ' world!')
+    assert text == 'hello world!'
+
+
+@chrome_only
+@sync
+async def test_Jeval_accepts_elementHandle(isolated_page):
+    """Test Page().Jeval() should accept ElementHandle object as args."""
+    page = isolated_page
+    await page.setContent('<section>hello</section><div> world</div>')
+    divHandle = await page.J('div')
+    text = await page.Jeval('section', "(e, div) => e.textContent + div.textContent", divHandle)
+    assert text == 'hello world'
+
+
+@chrome_only
+@sync
+async def test_Jeval_throws_exc_if_no_element(isolated_page):
+    """Test Page().Jeval() should throw error if no element is found."""
+    page = isolated_page
+    with pytest.raises(ElementHandleError) as exc:
+        await page.Jeval('section', "e => e.id")
+    assert 'Error: failed to find element matching selector "section"' in str(exc)
+
+
+@chrome_only
+@sync
+async def test_page_JJeval_executes_js_func(isolated_page):
+    """Test Page().JJeval() method executes JS function
+    with all elements which matches ``selector``.
+
+    """
+    page = isolated_page
+    await page.setContent('<div>hello</div><div>beautiful</div><div>world!</div>')
+    divsCount_jjeval = await page.JJeval('div', "divs => divs.length")
+    divsCount_query = await page.querySelectorAllEval('div', "divs => divs.length")
+    assert divsCount_jjeval == divsCount_query == 3
+
+
+@chrome_only
+@sync
+async def test_page_J_queries_element(isolated_page):
+    """Test Page().J() method should query existing element."""
+    page = isolated_page
+    await page.setContent('<section>test</section>')
+    element = await page.J('section')
+    assert element
+
+
+@chrome_only
+@sync
+async def test_J_returns_none_if_no_element(isolated_page):
+    """Test Page().J() method should return null for non-existing element."""
+    page = isolated_page
+    element = await page.J('non-existing-element')
+    assert not element
```

Comment:
This would be a little clearer as to what is expected:
```suggestion
    assert element is None
```
Otherwise a falsy value (empty str, etc) would pass this test

Reply:
fixed.